### ISSUE TITLE
Porting scimax's ob-ipython hydra

### ISSUE THREAD

**Issue by UndeadKernel** [dzop/emacs-jupyter#44]
The ob-ipython version in scimax has a very nice hydra that allows us to interact with an org file as if it was a jupyter notebook. The hydra provides functions to restart the kernel, add new source blocks, split a block, clone a block, run all blocks up to a point and many others. The hydra's doc-string looks like this:
```lisp
"
        Execute                   Navigate       Edit             Misc
----------------------------------------------------------------------
    _<return>_: current           _i_: previous  _w_: move up     _/_: inspect
  _S-<return>_: current to next   _k_: next      _s_: move down   _l_: clear result
_S-M-<return>_: to point          _g_: visible   _x_: kill        _L_: clear all
  _s-<return>_: Restart/block     _G_: any       _n_: copy        _._: complete
_M-s-<return>_: Restart/to point  ^ ^            _c_: clone
  _H-<return>_: Restart/buffer    ^ ^            _m_: merge
           _r_: Goto repl         ^ ^            _-_: split
           ^ ^                    ^ ^            _+_: insert above
           ^ ^                    ^ ^            _=_: insert below
           ^ ^                    ^ ^            _h_: header"
```

I have ported all these functions to work with `emacs-jupyter`.
@dzop, would you be interested in a pull request to add all these functions to `jupyter-org-client`?

**Comment by dzop** [comment]
Sure, but is there a way to do it so that there is no dependency on `hydra`? For example, by loading the hydra only if the `hydra` package is already present? Maybe use something like `(featurep 'hydra)`, `(require 'hydra nil t)`, or some kind of `hydra-loaded-hook`.

**Comment by dzop** [comment]
Closed by #45.
